plugins/woocommerce/client/legacy/js/admin/woocommerce_admin.js

```diff
@@ -326,11 +326,14 @@
 				function () {
 					var global_unique_id = $( this ).val();
 					$( this ).val(
-						global_unique_id.replace( /[^0-9\-]/g, '' )
+						global_unique_id
+							.replace( /[^0-9\-]/g, '' )
+							.replace( /^-+|-+$/g, '' )
 					);
-					$( document.body ).triggerHandler(
-						'wc_remove_error_tip',
-						[ $( this ), 'i18n_global_unique_id_error' ]
+
+					$( document.body ).triggerHandler( 
+						'wc_remove_error_tip', 
+						[ $( this ), 'i18n_global_unique_id_error']
```

Comment:
Heads-up that these lines got some unnecessary space changes:

```suggestion
					$( document.body ).triggerHandler(
						'wc_remove_error_tip',
						[ $( this ), 'i18n_global_unique_id_error' ]
```

Reply:
@Aljullu Thanks for the review. I have made the necessary changes to the PR. :)